plugins/net_plugin/net_plugin.cpp

```diff
@@ -1614,20 +1615,38 @@ namespace eosio {
       return true;
    }
 
-   void sync_manager::sync_reset_lib_num(const connection_ptr& c) {
+   void sync_manager::sync_reset_lib_num(const connection_ptr& c, bool closing) {
       std::unique_lock<std::mutex> g( sync_mtx );
       if( sync_state == in_sync ) {
          sync_source.reset();
       }
       if( !c ) return;
-      if( c->current() ) {
+      if( !closing && c->current() ) {
          std::lock_guard<std::mutex> g_conn( c->conn_mtx );
          if( c->last_handshake_recv.last_irreversible_block_num > sync_known_lib_num ) {
             sync_known_lib_num = c->last_handshake_recv.last_irreversible_block_num;
          }
-      } else if( c == sync_source ) {
-         sync_last_requested_num = 0;
-         request_next_chunk( std::move(g) );
+      } else if ( closing ) {
+         // Closing connection, therefore its view of LIB can no longer be considered as we will no longer be connected.
+         // Determine current LIB of remaining peers as our sync_known_lib_num.
+         uint32_t highest_lib_num = 0;
+         for_each_block_connection( [&highest_lib_num]( const auto& cc ) {
+            std::lock_guard<std::mutex> g_conn( cc->conn_mtx );
+            if( cc->current() && cc->last_handshake_recv.last_irreversible_block_num > highest_lib_num ) {
+               highest_lib_num = cc->last_handshake_recv.last_irreversible_block_num;
+            }
+            return true;
+         } );
+         sync_known_lib_num = highest_lib_num;
+
+         // if closing the connection we are currently syncing from, then reset our last requested and next expected.
+         if( c == sync_source ) {
+            sync_last_requested_num = 0;
+            uint32_t head_blk_num = 0;
+            std::tie( std::ignore, head_blk_num, std::ignore, std::ignore, std::ignore, std::ignore ) = my_impl->get_chain_info();
+            sync_next_expected_num = head_blk_num + 1;
+            request_next_chunk( std::move(g) );
+         }
```

Comment:
so technically there is a hypothetical `else` where `c` is not current and we are not closing.
Is this your intention to do nothing on that condition?
It looks like previously we were requesting next chunk anyway.

Reply:
Previously `closing` was implicit. `c` would not be `current()` if `closing==true`.

Reply:
ok, in that case can you also add `else {}` with warning/error log that it supposed to be unreachable. Just in case if this condition will arise (e.g. as a result of future changes) we will see this in logs

Reply:
Added log statement